cdapython/Q.py

```diff
@@ -591,20 +613,32 @@ def run(
         Returns:
             Optional[Result]: _description_
         """
+        dry_run_current = False
+
+        if host is None:
+            host = self._config.host
+        if table is None:
+            table = self._config.table
+
+        if dry_run is True:
+            dry_run = False
+            dry_run_current = True
+
         cda_client_obj: ApiClient = ApiClient(
             configuration=CdaConfiguration(host=host, verify=verify, verbose=verbose)
         )
         PAGEOFFSET = 0  # this variable is used as offset for the query function
+
         version, table = check_version_and_table(version, table)
 
         if include is not None:
             self.query = Q.__select(self, fields=include).query
 
         if limit is not None:
-            self.query = Q.__limit(self, limit)
+            self.query = Q.LIMIT(self, number=limit).query
 
         if offset > 0:
-            self.query = Q.__offset(self, offset)
+            self.query = Q.__offset(self, offset).query
```

Comment:
is there a reason not to use Q.OFFSET here like you did with LIMIT above?

Reply:
thanks for the catch I just made the OFFSET method to test thanks for that